lib/perl/Genome/InstrumentData/AlignmentResult/Command/RecreatePerLaneBam.pm

```diff
@@ -32,6 +32,11 @@ class Genome::InstrumentData::AlignmentResult::Command::RecreatePerLaneBam {
             doc => 'Picard version to replace bam header',
             valid_values => [Genome::Model::Tools::Picard->available_picard_versions],
         },
+        bam_header => {
```

Comment:
I agree. I don't know the context this is going to be called in, but it seems reasonable that the header location logic live somewhere else.